backend/services/product/internal/product/product.go

```diff
@@ -0,0 +1,116 @@
+package product
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/manaaan/ekolivs-oms/pkg/env"
+	"github.com/manaaan/ekolivs-oms/pkg/zettle"
+	"github.com/manaaan/ekolivs-oms/product/api"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+type Service struct {
+	zettleService *zettle.Service
+	storeService  *Store
+}
+
+func New() (*Service, error) {
+	zettleService, err := zettle.New(zettle.ServiceNewParams{
+		ClientId: env.Required("ZETTLE_ORG_UUID"),
+		ApiKey:   env.Required("ZETTLE_API_KEY"),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return &Service{
+		zettleService: zettleService,
+		storeService:  &Store{},
+	}, nil
+}
+
+func (s Service) GetProducts() ([]*api.Product, error) {
+	zettleProducts, err := s.zettleService.GetProducts()
+	if err != nil {
+		return nil, err
+	}
+
+	storeProducts, err := s.storeService.GetProducts()
+	fmt.Println(storeProducts)
```

Comment:
Should probably have a follow up issue here on what we'll plan to do/add in terms of information.

We would want to have the Supplier attached to the product.
This would need to be managed somewhere as well, or we base it on automated mapping based on the name formatting we have in Zettle.

Reply:
Let's discuss next time what we want to store. I forgot what we decided before. My idea would be to store everything in our database and update our information with the [Zettle Pusher API](https://developer.zettle.com/docs/api/pusher/api-reference-md), listening to the event `ProductUpdated`. Not sure if this was in scope of V1.

Is your naming consistent? Otherwise we might start off from the automated mapping and then create a manual process of assigning products to suppliers via the UI. It should also handle the creation of new suppliers.
In the future, we also want to store base data for suppliers, not just the name. So an automated mapping will not suffice.

Reply:
All products should be stored in the database. There should be one endpoint to refetch all products. There should be a cron that updates product information once a week. The endpoint to refetch should not run more than once every minute. 
The products endpoint should be paginated and only return products from our database.